### PR TITLE
gen config.toml, overrides tmint default

### PR DESCRIPTION
**Pull request by zramsay**
Let's you set config. Otherwise `tendermint node` makes a default that needs to be changed & chain restarted.

### CODE REVIEW

**Comment by ebuchman** [comment]
Hey I'd like this to also fix the folder writing so that when you do `mintgen random 5` you get 5 folders each with the same `genesis.json` and `config.toml` and each with their own `priv_validator.json`. This will make it super easy to start test nets by running `mintgen random 5` and then pushing each folder to a remote

**Comment by ebuchman** [comment]
actually, would be really great if we could also get some additional options, like passing in the pubkeys we want to use for validators, and maybe passing a csv file containing addresses to include in the genesis with their starting balance and permissions. This would basically address and enhance https://github.com/eris-ltd/mint-client/issues/12

**Comment by zramsay** [comment]
by "passing in the pubkeys we want to use for validators", is there something meant over and above what #12 already is?

**Comment by ebuchman** [comment]
no not really, though it might be nice if there was an option to pass validator pubkeys in from a simple text file in case there are many, so we don't have to list from command line.

**Comment by zramsay** [comment]
should be ready to go. `--csv` is only missing autopop for params, see #16

File: mintx/core.go [hunk 1]
```diff
@@ -254,9 +254,11 @@ func coreBond(nodeAddr, pubkey, unbondAddr, amtS, nonceS string) (*types.BondTx,
 	if err != nil {
 		return nil, err
 	}
-
+	var pubKey account.PubKeyEd25519
 	if unbondAddr == "" {
-		return nil, fmt.Errorf("Unbond address must be given with --unbond-to flag")
+		pkb, _ := hex.DecodeString(pubkey) //err checked in checkCommon()
+		copy(pubKey[:], pkb)
+		unbondAddr = string(pubKey.Address())
```
Comment (ebuchman) [inline mintx/core.go]: we don't care about unbondAddr (the string), just unbondAddrBytes. Also the next couple lines of code should be in an `else` block

**Comment by ebuchman** [comment]
also gonna need a rebase because i just pushed some changes to mintx

File: mintgen/cli.go [hunk 2]
```diff
@@ -146,17 +148,149 @@ func cliRandom(cmd *cobra.Command, args []string) {
 		IfExit(os.MkdirAll(DirFlag, 0700))
 	}
 	// XXX: we're gonna write to it even if it already exists
+	IfExit(ioutil.WriteFile(path.Join(DirFlag, "config.toml"), []byte(defaultConfig), 0644))
 	IfExit(ioutil.WriteFile(path.Join(DirFlag, "genesis.json"), genesisBytes, 0644))
+	fmt.Println("dirflag", DirFlag)
```
Comment (ebuchman) [inline mintgen/cli.go]: cleanup your print statements

File: mintgen/cli.go [hunk 3]
```diff
@@ -146,17 +148,149 @@ func cliRandom(cmd *cobra.Command, args []string) {
...
 	for i, v := range validators {
 		buf, n = new(bytes.Buffer), new(int64)
 		wire.WriteJSON(v, buf, n, &err)
 		IfExit(err)
 		valBytes := buf.Bytes()
 		if len(validators) > 1 {
-			IfExit(ioutil.WriteFile(path.Join(DirFlag, fmt.Sprintf("priv_validator_%d.json", i)), valBytes, 0600))
+			mulDir := DirFlag + "_" + strconv.Itoa(i)
```
Comment (ebuchman) [inline mintgen/cli.go]: fmt.Sprintf("%s_%d", DirFlag, i)

File: mintgen/cli.go [hunk 4]
```diff
@@ -146,17 +148,149 @@ func cliRandom(cmd *cobra.Command, args []string) {
...
+			fmt.Println("muldir", mulDir)
+			fmt.Println("i", i)
+			IfExit(os.MkdirAll(mulDir, 0700))
+			IfExit(ioutil.WriteFile(path.Join(mulDir, fmt.Sprintf("priv_validator_%d.json", i)), valBytes, 0600))
+			IfExit(ioutil.WriteFile(path.Join(mulDir, "config.toml"), []byte(defaultConfig), 0644))
```
Comment (ebuchman) [inline mintgen/cli.go]: the whole point was for these to just be `priv_validator.json`, not `priv_validator_7.json` ....

File: mintgen/cli.go [hunk 5]
```diff
@@ -146,17 +148,149 @@ func cliRandom(cmd *cobra.Command, args []string) {
...
+		Exit(fmt.Errorf("Enter one or more pub keys or path to .csv"))
+	}
+
+	var pubkeys []string
+	var amts []string
+	var names []string
+	//	var perms []string
+
+	if CsvPathFlag != "" {
+		//TODO figure out perms
```
Comment (ebuchman) [inline mintgen/cli.go]: nice

**Comment by ebuchman** [comment]
we should add tests for the csvs

File: mintgen/cli.go [hunk 6]
```diff
@@ -146,17 +148,147 @@ func cliRandom(cmd *cobra.Command, args []string) {
...
+			IfExit(os.MkdirAll(mulDir, 0700))
+			IfExit(ioutil.WriteFile(path.Join(mulDir, "priv_validator.json"), valBytes, 0600))
+			IfExit(ioutil.WriteFile(path.Join(mulDir, "config.toml"), []byte(defaultConfig), 0644))
+			IfExit(ioutil.WriteFile(path.Join(mulDir, "genesis.json"), genesisBytes, 0644))
 		} else {
 			IfExit(ioutil.WriteFile(path.Join(DirFlag, "priv_validator.json"), valBytes, 0600))
 		}
 	}
-	fmt.Printf("genesis.json and priv_validator_X.json files saved in %s\n", DirFlag)
+	fmt.Printf("config.toml, genesis.json and priv_validator_X.json files saved in %s\n", DirFlag)
```
Comment (ebuchman) [inline mintgen/cli.go]: :P

File: mintgen/cli.go [hunk 7]
```diff
@@ -146,17 +148,147 @@ func cliRandom(cmd *cobra.Command, args []string) {
...
+	var amts []string
+	var names []string
+	//	var perms []string
+
+	if CsvPathFlag != "" {
+		//TODO figure out perms
+		pubkeys, amts, names, _ = parseCsv(CsvPathFlag)
+
+	} else {
+		pubkeys = strings.Split(PubkeyFlag, " ")
```
Comment (ebuchman) [inline mintgen/cli.go]: I don't think you can split flags on spaces. use commas.

File: mintgen/cli.go [hunk 8]
```diff
@@ -146,17 +148,147 @@ func cliRandom(cmd *cobra.Command, args []string) {
...
+		pubKeyBytes[i], err = hex.DecodeString(k)
+		if err != nil {
+			Exit(fmt.Errorf("Pubkey (%s) is invalid hex: %v", k, err))
+		}
+	}
+
+	amt := make([]int64, len(amts))
+	for i, a := range amts {
+		var err error
+		amt[i], err = strconv.ParseInt(a, 10, 64)
```
Comment (ebuchman) [inline mintgen/cli.go]: these are gonna be 0 if theres no csv

File: mintgen/cli.go [hunk 9]
```diff
@@ -146,17 +148,147 @@ func cliRandom(cmd *cobra.Command, args []string) {
...
+	unbAmt := int64(1) << 50
+
+	for i, kb := range pubKeyBytes {
+		copy(pubKey[:], kb)
+		addr := pubKey.Address()
+
+		genDoc.Accounts[i] = state.GenesisAccount{
+			Address: addr,
+			Amount:  amt[i],
+			Name:    names[i],
```
Comment (ebuchman) [inline mintgen/cli.go]: you never made sure Names is filled in (ie if not csv). this panics

File: mintgen/main.go [hunk 10]
```diff
@@ -24,13 +27,24 @@ func main() {
 		Long:  "Create <N> keys and a genesis.json with corresponding validators and chain_id <name>",
 		Run:   cliRandom,
 	}
-	randomCmd.Flags().StringVarP(&DirFlag, "dir", "d", "", "Directory to save genesis and priv_validators in")
+	randomCmd.Flags().StringVarP(&DirFlag, "dir", "d", "", "Directory to save genesis and priv_validators in. Default is ~/.eris/data/<chain_id>")
```
Comment (ebuchman) [inline mintgen/main.go]: I'm thinking random should also take a "seed_node" flag that gets written into the config. parsing the config and replacing the seed_node might be a bit annoying but its basically essential for proper deploys

Comment (zramsay) [inline mintgen/main.go]: perfect...that can go in the shell script too

**Pull request merged**